fastparse/shared/src/main/scala/fastparse/Api.scala

```diff
@@ -12,6 +12,8 @@ trait Api{
 
   val Result = core.Result
   type Result[+T] = core.Result[T]
+  val Failure = core.Result.Failure
+  val Success = core.Result.Success
```

Comment:
Perhaps we could renamed `Result` into `Parsed` for even better disambiguation? `Parsed.Success` and `Parsed.Failure` seems just about perfect